### PR TITLE
Add SubscribeRule and SubscribeRulesUpdate

### PR DESCRIPTION
**Pull request by innerverse**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: src/main/java/com/twilio/type/SubscribeRule.java [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+        this.type = null;
+        this.all = null;
+        this.kind = null;
+        this.priority = null;
+        this.publisher = null;
+        this.track = null;
+    }
+
+    @JsonIgnore
+    public boolean isValid() {
```
Comment (childish-sambino) [inline src/main/java/com/twilio/type/SubscribeRule.java]: Logic looks correct to me, but really needs tests to help with future maintenance.

File: src/main/java/com/twilio/type/SubscribeRule.java [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+            @JsonProperty("all") final Boolean all,
+            @JsonProperty("publisher") final String publisher,
+            @JsonProperty("track") final String track,
+            @JsonProperty("kind") final Kind kind,
+            @JsonProperty("priority") final Priority priority) {
+        this.type = type;
+        this.all = all;
+        this.kind = kind;
+        this.priority = priority;
+        this.publisher = publisher;
```
Comment (childish-sambino) [inline src/main/java/com/twilio/type/SubscribeRule.java]: nit: ordering is different.

Comment (innerverse) [reply src/main/java/com/twilio/type/SubscribeRule.java]: Fixed, thanks!

File: src/main/java/com/twilio/type/SubscribeRule.java [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+        private final String value;
+
+        Kind(final String value) {
+            this.value = value;
+        }
+
+        @JsonCreator
+        public static Type forValue(final String value) {
+            return Promoter.enumFromString(value, Type.values());
```
Comment (childish-sambino) [inline src/main/java/com/twilio/type/SubscribeRule.java]: `s/Type/Kind/`

File: src/main/java/com/twilio/type/SubscribeRule.java [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+
+        private final String value;
+
+        Priority(final String value) {
+            this.value = value;
+        }
+
+        @JsonCreator
+        public static Type forValue(final String value) {
+            return Promoter.enumFromString(value, Type.values());
```
Comment (childish-sambino) [inline src/main/java/com/twilio/type/SubscribeRule.java]: `s/Type/Priority/`

**Pull request merged**